Applications/Spire/Source/Login/LoginWindow.cpp

```diff
@@ -38,19 +54,17 @@ LoginWindow::LoginWindow(const std::string& version, QWidget* parent)
   title_bar_layout->setContentsMargins({});
   title_bar_layout->setSpacing(0);
   title_bar_layout->addStretch(352);
-  auto exit_button_style = IconButton::Style();
-  exit_button_style.m_default_color = "#E2E0FF";
-  exit_button_style.m_hover_color = "#E63F45";
-  exit_button_style.m_blur_color = "#BAB3D9";
-  exit_button_style.m_hover_background_color = "#321471";
-  m_exit_button = new IconButton(imageFromSvg(":/Icons/close.svg",
-    BUTTON_SIZE()), exit_button_style, this);
-  m_exit_button->setFixedSize(BUTTON_SIZE());
-  m_exit_button->setFocusPolicy(Qt::NoFocus);
-  m_exit_button->installEventFilter(this);
-  connect(m_exit_button, &IconButton::clicked, [=] { window()->close(); });
-  title_bar_layout->addWidget(m_exit_button);
-  title_bar_layout->setStretchFactor(m_exit_button, 32);
+  m_close_button = make_icon_button(imageFromSvg(":/Icons/close.svg",
+    BUTTON_SIZE()), this);
+  set_style(*m_close_button, CLOSE_BUTTON_STYLE());
+  m_close_button->setFixedSize(BUTTON_SIZE());
+  m_close_button->setFocusPolicy(Qt::NoFocus);
+  m_close_button->installEventFilter(this);
+  m_close_button->connect_clicked_signal([=] {
+    window()->close();
+  });
+  title_bar_layout->addWidget(m_close_button);
+  title_bar_layout->setStretchFactor(m_close_button, 32);
```

Comment:
What is this ```32``` used for?

Reply:
This comes from before I started using fixed sizes and spacing for widgets in layouts. By setting stretches the layout will automatically resize the widgets based on these stretch values, which is contained by the fixed size of the window. The layout won't have any effect on updating this window, but if you want I can update it to use fixed size widgets and spacing to lay out the window contents.

Reply:
Okay keep it as is. As long as this scales properly based on resolution/DPI then it's fine.